sechub-integrationtest/src/main/java/com/mercedesbenz/sechub/integrationtest/internal/SecHubClientExecutor.java

```diff
@@ -342,13 +343,28 @@ private File resolveExistingExecutableAndAppendAdditionalCommands() {
 
         if (sechubClientBinaryPath == null) {
             String parentFolder = "sechub-cli/build/go/platform/"; // when not set we use build location
-            String sechubExeName = null;
-            if (TestUtil.isWindows()) {
-                sechubExeName = "sechub.exe";
-                parentFolder += "windows-386";
+            String sechubExeName = "sechub";
+
+            String osArch = SystemUtils.OS_ARCH;
+            boolean is64 = osArch.contains("64");
+            boolean isArm = osArch.contains("arm") || osArch.contains("aarch");
+
+            if (SystemUtils.IS_OS_WINDOWS && !isArm) {
+                sechubExeName += ".exe";
+                parentFolder += "windows";
+                parentFolder += (is64) ? "-amd64" : "-386";
+            } else if (SystemUtils.IS_OS_LINUX) {
+                parentFolder += "linux";
+                if (isArm) {
+                    parentFolder += (is64) ? "-arm64" : "-arm";
+                } else {
+                    parentFolder += (is64) ? "-amd64" : "-386";
+                }
+            } else if (SystemUtils.IS_OS_MAC_OSX && is64) {
+                parentFolder += "darwin";
+                parentFolder += (isArm) ? "-arm64" : "-amd64";
```

Comment:
(very minor)

I was thinking about making this more readable by extracting this `is64`, `isArm` logic somewhere. It's more code, but seems to be slightly more readable(?). I'm not sure, so feel free to use it, if you like it:

```java
            String parentFolderSuffix;
            if(is64) {
                if(isArm) {
                    parentFolderSuffix = "-arm64";
                } else {
                    parentFolderSuffix = "-amd64";
                }
            } else {
                if(isArm) {
                    parentFolderSuffix = "-arm";
                } else {
                    parentFolderSuffix = "-386";
                }
            }
            if (SystemUtils.IS_OS_WINDOWS && !isArm) {
                sechubExeName += ".exe";
                parentFolder += "windows";
            } else if (SystemUtils.IS_OS_LINUX) {
                parentFolder += "linux";
            } else if (SystemUtils.IS_OS_MAC_OSX && is64) {
                parentFolder += "darwin";
            } else {
                throw new RuntimeException(
                        "Unknown OS (" + SystemUtils.OS_NAME + ") or processor architecture (" + osArch + ")"
                );
            }
            parentFolder += parentFolderSuffix;
```

Reply:
Tbh, I think I find the current version more readable, but I appreciate the effort :D